tools/topology/topology2/cavs/cavs-sdw-aic.conf

```diff
@@ -0,0 +1,287 @@
+<searchdir:cavs>
+<searchdir:include>
+<searchdir:include/common>
+<searchdir:include/components>
+<searchdir:include/dais>
+<searchdir:include/pipelines/cavs>
+<searchdir:cavs/platform/intel>
+
+<vendor-token.conf>
+<pdm_config.conf>
+<tokens.conf>
+<virtual.conf>
+<host-copier-gain-mixin-playback.conf>
+<mixout-gain-dai-copier-playback.conf>
+<deepbuffer-playback.conf>
+<passthrough-playback.conf>
+<passthrough-capture.conf>
+<passthrough-be.conf>
+<data.conf>
+<pcm.conf>
+<pcm_caps.conf>
+<fe_dai.conf>
+<alh.conf>
+<hda.conf>
+<dmic.conf>
+<hw_config.conf>
+<manifest.conf>
+<route.conf>
+<cavs/common_definitions.conf>
+<copier.conf>
+<pipeline.conf>
+<dai.conf>
+<host.conf>
+<dmic-default.conf>
+<hdmi-default.conf>
+
+Define {
+	DMIC_IO_CLK 38400000
+	NUM_DMICS 0
+	DMIC_DRIVER_VERSION 3
+	# override DMIC default definitions
+	PDM1_MIC_A_ENABLE 1
+	PDM1_MIC_B_ENABLE 1
+	DMIC0_HOST_PIPELINE_ID 13
+	DMIC0_DAI_PIPELINE_ID 14
+	DMIC0_HOST_PIPELINE_SINK 'copier.host.13.1'
+	DMIC0_DAI_PIPELINE_SRC 'copier.DMIC.14.1'
+	DMIC0_NAME 'dmic01'
+	DMIC0_ID 4
+	DMIC1_ID 5
+	DMIC0_PCM_CAPS 'Passthrough Capture 13'
+	DMIC0_PIPELINE_STREAM_NAME 'copier.DMIC.14.1'
+	USE_CHAIN_DMA	"false"
+}
+
+# include DMIC config if needed.
+IncludeByKey.NUM_DMICS {
+"[1-4]" "platform/intel/dmic-generic.conf"
+}
+
+Define {
+	NUM_HDMIS 0
+}
+
+# include HDMI config if needed.
+IncludeByKey.NUM_HDMIS {
+"[3-4]" "platform/intel/hdmi-generic.conf"
+}
+
+#
+# List of all DAIs
+#SDW2-Headset codec, SDW0-Spk amp
+#
+Object.Dai {
+	ALH."514" {
```

Comment:
it's a mess. We need to compute the ALH identifiers from the position of the links and PDIs, not hard-code like this.

It seems like with topology2 everyone forgot about the need to configure multiple hardware layouts as we did for topology1.

Reply:
and it's again completely wrong to create a new topology per card, we've got to reuse reuse and reuse.

Reply:
Hi @ranj063  and @plbossart ,
Sorry for the delay in response, understood your comments. Had a discussion with @bardliao , and he is already working on this in https://github.com/thesofproject/sof/pull/6549/commits. Hence, to avoid redundancy, closing this PR and will work with Bard.
Thanks.